### PR TITLE
[Infineon] Not to increment RebootCount if wakes from sleep state

### PR DESCRIPTION
**Pull request by AlvinHsiao**
### Problem
RebootCount would increment if wakes from sleep state.

### Change overview
Not to increment RebootCount if wakes from sleep state.

### Testing
Tested manually with chip-tool.

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/Infineon/CYW30739/ConfigurationManagerImpl.cpp [hunk 1]
```diff
@@ -51,9 +52,11 @@ CHIP_ERROR ConfigurationManagerImpl::Init()
     {
         err = GetRebootCount(rebootCount);
         SuccessOrExit(err);
-
-        err = StoreRebootCount(rebootCount + 1);
-        SuccessOrExit(err);
+        if (wiced_sleep_get_boot_mode() != WICED_SLEEP_FAST_BOOT)
```
[the remark this answers is not in the document]
Comment (AlvinHsiao) [reply src/platform/Infineon/CYW30739/ConfigurationManagerImpl.cpp:55]: We just followed the Matter spec and not to increment RebootCount from sleep state.
image

**Comment by bzbarsky-apple** [comment]
If it's a low-power enough state that it's lost volatile memory, does that count as "wakes up from a sleep state", is the question?  Assuming that's the situation here, of course.

File: src/platform/Infineon/CYW30739/ConfigurationManagerImpl.cpp (hunk 1 above)
Comment (bzbarsky-apple) [inline src/platform/Infineon/CYW30739/ConfigurationManagerImpl.cpp:55]: Another question: when is `PlatformMgrImpl().GetStartTime()` being set, for purposes of GetUpTime?  That one:

1) Needs to match the reboot count bit
2) Needs to match the time base for events.

Comment (AlvinHsiao) [reply src/platform/Infineon/CYW30739/ConfigurationManagerImpl.cpp:55]: For the first question, we will save data into retention RAM before sleep. And it count as "wakes up from a sleep state".
For the second question, I think PlatformMgrImpl().GetStartTime() is for Linux specific, and we didn't implement it on our platform.

Comment (bzbarsky-apple) [reply src/platform/Infineon/CYW30739/ConfigurationManagerImpl.cpp:55]: @AlvinHsiao Which data is being saved into retention RAM?

I see that I just misread things and GetStartTime is relevant to `Infineon/PSOC6`, not `Infineon/CYW30739`, and that `Infineon/CYW30739` just does not implement the UpTime attribute at all.  Is that a correct description of the current situation?

And your're shutting down the Matter stack before entering the sleep state and restarting it when you exit?

Comment (HuangCC-IFX) [reply src/platform/Infineon/CYW30739/ConfigurationManagerImpl.cpp:55]: @bzbarsky-apple Yes, this is our temporary workaround to support sleepy end device.

Comment (bzbarsky-apple) [reply src/platform/Infineon/CYW30739/ConfigurationManagerImpl.cpp:55]: @HuangCC-IFX That doesn't quite answer my question around which data is persisted.  For example, when you enter this sleep state, do you clear out the queued events in the Matter stack?  Do you issue a new StartUp event when coming out and a ShutDown event when entering?

**Comment by andy31415** [comment]
PR not updated for a long time, has CI failures and unresolved comments. Assuming stale and closing.

**Pull request closed without merging**